File: week2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include<string.h>
#include<ctype.h>
int shift(char c);

int main(int argc, string argv[])
{
    if (argc != 2)
        {
            printf("Usage: %s keyword\n", argv[0]);
            return 1;
        }

    for (int i = 0, n = strlen(argv[1]); i < n; i++)
        {
            if (isalpha(argv[1][i])) // якщо символ із рядка є буквою, то вірно
                {
                    true;
                }
            else
                {
                    printf("Usage: %s keyword\n", argv[0]);
                    return 2;
                }
        } // створення змінних типу і потрібних нижче
    int parol = strlen(argv[1]),j = 0, key;
    //отримання тексту для шифрування від користувача
    string plain = get_string("plaintext: ");
    printf("ciphertext: ");

    for( int i = 0, n = strlen(plain); i < n; i++)
        {// процес шифрування. для маленьких і великих букв окремо.
            char text = plain[i];
            if(isalpha(text))
               {
                    key = shift(argv[1][j]);
                    if(islower(text))
                        {
                            printf("%c", (text - 'a' + key)%26+'a');
                        }
                    else
                        {
                            printf("%c", (text - 'A' + key)%26+'A');
                        }
                    j = (j+1)%parol;
                }
            else
                {
                    printf("%c", text);
                }
        }
    printf("\n");
    return 0;
}

// нова функція
int shift(char c)
{// присвоюємо цифрове значення для букв від 0=А.
    char big = toupper(c);
    int shift = (int) big -'A';
    return shift;
}

File: week6/vigenere.py
from cs50 import get_string
from sys import argv
import sys


def main():
    # перевірка чи є всі символи ключа користувача буквами
    if len(argv) is not 2 or argv[1].isalpha() == False:
        print(f"Usage:{argv[0]} key")
        sys.exit(1)

    text = get_string("plaintext: ")  # текст користувача для шифру
    print("ciphertext: ", end="")

    cipher(text)
    print()


def cipher(text):
    c = 0
    # шифруємо текст
    for i in range(len(text)):
        if text[i].isalpha():

            key = ord(argv[1][c].upper()) - 65  # ключ

            if text[i].islower():
                print(f"{ chr((ord(text[i]) - 97 + key )%26 + 97)}", end="")
            else:
                print(f"{ chr((ord(text[i]) - 65 + key)%26 + 65)}", end="")

            c = (c+1) % len(argv[1])  # переходимо на наступну букву ключа

        else:
            print(f"{text[i]}", end="")


if __name__ == "__main__":
    main()